File: gitnexus/test/fixtures/lang-resolution/java-qualified-base/src/app/Service.java
package app;

// Qualified-generic bases: `extends app.base.Box<String>` (generic_type wrapping
// a scoped_type_identifier) and `implements app.base.IFoo<String>` (in a
// type_list). Both resolve by their trailing simple name (Box / IFoo).
public class Service extends app.base.Box<String> implements app.base.IFoo<String> {
    public void foo(String t) {}
}

File: gitnexus/test/fixtures/lang-resolution/typescript-qualified-base/src/Service.ts
import * as ns from './base';

// Qualified-generic bases: `extends ns.Box<string>` (extends_clause value is a
// member_expression, type_arguments a sibling) and `implements ns.IFoo<string>`
// (implements_clause -> generic_type wrapping a nested_type_identifier). Both
// resolve by their trailing simple name (Box / IFoo).
export class Service extends ns.Box<string> implements ns.IFoo<string> {
  foo(t: string): void {}
}

// Qualified non-generic bases: `extends ns.Base` (member_expression) and
// `implements ns.IBar` (nested_type_identifier).
export class Plain extends ns.Base implements ns.IBar {
  bar(): void {}
}
